Fix passing `onLoad` overwrites, and call `onLoad` inside `onImageLoad`

johnsonsu opened the pull request:
Issue: Currently, there is no way to retrieve the `onLoad` event object with this library and passing an `onLoad` prop to `LazyLoadImage` will overwrite the handler on line 47 of `LazyLoadImage.jsx` causing `afterLoad` to never get called.

**Description**
This PR fixes passing `onLoad` to `LazyLoadImage` causing `afterLoad()` to never fire:
- Fix `onLoad` getting overwritten by props which causes `onImageLoad()` to never get called.
- preserve the functionality of `onLoad` by adding a callback.

Aljullu commented:
Thanks for your contribution, @johnsonsu! I agree that it would be a good idea to pass the event object. However, it's not clear to me whether we need to introduce a new prop for this. Could we simply use `afterLoad` instead (maybe renaming it to `onLoad`)? Would that work for your use-case?

johnsonsu commented:
Hi @Aljullu, hope you are doing well! Yes, that will work for my use case. I can update my code to include the event in the `afterLoad` callback like:
```jsx
return e => {
    this.props.afterLoad(e);
    ...
```
However, I think it's still a good idea to preserve the `onLoad` for consistency because your library carries over all other props of `img` and having `onLoad` as an exception is confusing. In fact, I created this PR because I was expecting `onLoad` to work just like a regular `img` but it didn't. What do you think?

Aljullu commented:
Good points, @johnsonsu. I'm thinking we should maybe simply rename the `afterLoad` to `onLoad` for the reasons you mention. And perhaps keep `afterLoad` as an alias of `onLoad` to keep backwards compatibility. Does it make sense to you? Are you up to giving it a spin? Ideally, we would update the docs as well.

johnsonsu commented:
> Good points, @johnsonsu. I'm thinking we should maybe simply rename the `afterLoad` to `onLoad` for the reasons you mention. And perhaps keep `afterLoad` as an alias of `onLoad` to keep backwards compatibility. Does it make sense to you? Are you up to giving it a spin? Ideally, we would update the docs as well.

I think that make sense. I will update the code and docs and push it to this PR in a bit.

johnsonsu commented:
@Aljullu This is ready for your review.

Aljullu commented on src/components/LazyLoadImage.jsx:
```diff
@@ -17,8 +17,9 @@ class LazyLoadImage extends React.Component {
-		return () => {
-			this.props.afterLoad();
+		return e => {
+			this.props.onLoad(e);
+			this.props.afterLoad(e);
```
What do you think about adding a comment here? Something along these lines:

```suggestion
			// We keep support for afterLoad for backwards compatibility,
			// but `onLoad` is the preferred prop.
			this.props.onLoad(e);
			this.props.afterLoad();
```

I would also not pass the event object to `afterLoad`, so we don't change the signature of that function.

Aljullu reviewed (commented):
Overall looks good, thanks again for working on this. I left one comment inline with a suggestion. Also, you will need to add `onLoad` to the `propTypes` and `defaultProps` of that file.

Besides that, it might be a good idea to [duplicate this test](https://github.com/Aljullu/react-lazy-load-image-component/blob/master/src/components/LazyLoadImage.spec.js#L70-L83) with the `onLoad` prop.

johnsonsu commented:
I think that's a good suggestion.

I've updated the code and labelled `afterLoad` as deprecated in the document and with comments.
I also updated the test to include both `onLoad` and `afterLoad` in the same test. This way the test can verify that passing `onLoad` no longer overwrites.

Aljullu reviewed (approved):
Awesome, this is looking good, @johnsonsu! Thanks a lot for your contribution! :raised_hands:
